CMakeLists.txt

```diff
@@ -20,10 +20,9 @@ set_pennylane_lightning_version("${CMAKE_SOURCE_DIR}/pennylane_lightning/_versio
 message(STATUS "pennylane_lightning version ${VERSION_STRING}")
 set(PROJECT_VERSION ${VERSION_STRING})
 
-set(CMAKE_CXX_STANDARD 11) # At least C++11 is required
+set(CMAKE_CXX_STANDARD 17) # At least C++17 is required
 find_package(OpenMP REQUIRED) # find OpenMP
```

Comment:
GateData.hpp was a carry=over from the googletest suite. I updated them, then decided to pull them out. You are completely correct that it can be pulled. I'll removed all the remants of the old testing suite.

Reply:
After a little thought, and since there are a few other places and libs to tackle the removal of, I'd rather move this to a separate story, and keep this PR to the templating.